odoo/addons/base/models/avatar_mixin.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from hashlib import sha512
+from base64 import b64encode
+
+from odoo import models, fields, api
+
+from odoo.tools import html_escape
+
+
+class AvatarMixin(models.AbstractModel):
+    _name = 'avatar.mixin'
+    _description = "Avatar Mixin"
+
+    # all image fields are base64 encoded and PIL-supported
+
+    avatar_1920 = fields.Image("Avatar", max_width=1920, max_height=1920, compute="_compute_avatar_1920", inverse="_set_avatar", readonly=False)
+
+    avatar_1024 = fields.Image("Avatar 1024", max_width=1024, max_height=1024, compute="_compute_avatar_1024", readonly=False)
+    avatar_512 = fields.Image("Avatar 512", max_width=512, max_height=512, compute="_compute_avatar_512", readonly=False)
+    avatar_256 = fields.Image("Avatar 256", max_width=256, max_height=256, compute="_compute_avatar_256", readonly=False)
+    avatar_128 = fields.Image("Avatar 128", max_width=128, max_height=128, compute="_compute_avatar_128", readonly=False)
+
+    # image fields are stored and hold the image data
+
+    image_1920 = fields.Image("Image", max_width=1920, max_height=1920, store=True)
+
+    # resized fields stored for performance
+    image_1024 = fields.Image("Image 1024", related="image_1920", max_width=1024, max_height=1024, store=True)
+    image_512 = fields.Image("Image 512", related="image_1920", max_width=512, max_height=512, store=True)
+    image_256 = fields.Image("Image 256", related="image_1920", max_width=256, max_height=256, store=True)
+    image_128 = fields.Image("Image 128", related="image_1920", max_width=128, max_height=128, store=True)
```

Comment:
Those fields are a direct copy of the same fields in the image mixin now actually, so it would be better to not have to duplicate them here. Ideally this mixin should inherit the other one, if that is even possible?